src/compressed_tensors/compressors/model_compressors/model_compressor.py

```diff
@@ -301,23 +308,41 @@ def decompress(self, model_path: str, model: Module):
         :param model: pytorch model to load decompressed weights into
         """
         model_path = get_safetensors_folder(model_path)
+        sparse_decompressed = False
+
         if self.sparsity_compressor is not None:
+            # Sparse decompression is applied on the model_path
             dense_gen = self.sparsity_compressor.decompress(model_path)
             self._replace_weights(dense_gen, model)
             setattr(model, SPARSITY_CONFIG_NAME, self.sparsity_compressor.config)
+            sparse_decompressed = True
 
         if self.quantization_compressor is not None:
-            names_to_scheme = apply_quantization_config(model, self.quantization_config)
-            load_pretrained_quantization(model, model_path)
+            # Temporarily set quantization status to FROZEN to prevent
+            # quantization during apply_quantization_config. This ensures
+            # that the dtypes of the weights are not unintentionally updated.
+            # The status is restored after quantization params are loaded.
+            with override_quantization_status(
+                self.quantization_config, QuantizationStatus.FROZEN
+            ):
```

Comment:
this is usually done outside of the ModelCompressor? It should always be frozen when decompressing, no?

Reply:
You are correct QuantizationStatus should be frozen when decompressing. However I think setting this should be owned by the model compressor?

Reply:
Yeah but we also set it at the end - I'm wondering if we should just be setting it at the beginning/if the context is really needed.

Reply:
At the end QuantizationStatus.FROZEN is applied at the module level, however during apply_quantization_config, the QuantizationStatus used is from the config, I added this because it's safer; we could just set quantization_config.quantization_status here if it does not have downstream effects, let me know if you think that'll be better